geomesa-features/geomesa-feature-confluent/src/main/scala/org/locationtech/geomesa/features/confluent/ConfluentFeatureSerializer.scala

```diff
@@ -0,0 +1,114 @@
+/***********************************************************************
+ * Copyright (c) 2013-2018 Commonwealth Computer Research, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, Version 2.0
+ * which accompanies this distribution and is available at
+ * http://www.opensource.org/licenses/apache2.0.php.
+ ***********************************************************************/
+
+package org.locationtech.geomesa.features.confluent
+
+import java.io.{InputStream, OutputStream}
+import java.net.URL
+import java.util.Date
+
+import com.typesafe.scalalogging.LazyLogging
+import org.locationtech.jts.geom.Geometry
+import org.locationtech.jts.io.WKTReader
+import io.confluent.kafka.schemaregistry.client.{CachedSchemaRegistryClient, SchemaRegistryClient}
+import io.confluent.kafka.serializers.KafkaAvroDeserializer
+import org.apache.avro.generic.GenericRecord
+import org.locationtech.geomesa.features.confluent.ConfluentFeatureSerializer._
+import org.locationtech.geomesa.features.SerializationOption.SerializationOption
+import org.locationtech.geomesa.features.{ScalaSimpleFeatureFactory, SimpleFeatureSerializer}
+import org.opengis.feature.simple.{SimpleFeature, SimpleFeatureType}
+
+import scala.collection.JavaConverters._
+import scala.util.control.NonFatal
+
+object ConfluentFeatureSerializer {
+  def builder(sft: SimpleFeatureType, schemaRegistryUrl: URL): Builder =
+    new Builder(sft, schemaRegistryUrl)
+
+  class Builder private [ConfluentFeatureSerializer] (sft: SimpleFeatureType, schemaRegistryUrl: URL)
+      extends SimpleFeatureSerializer.Builder[Builder] {
+
+    override def build(): ConfluentFeatureSerializer =
+      new ConfluentFeatureSerializer(sft,
+                                     new CachedSchemaRegistryClient(schemaRegistryUrl.toExternalForm, 100),
+                                     options.toSet)
+  }
+  val geomAttributeName = "_geom"
+  val dateAttributeName = "_date"
+}
+
+class ConfluentFeatureSerializer(sft: SimpleFeatureType,
+                                 schemaRegistryClient: SchemaRegistryClient,
+                                 val options: Set[SerializationOption] = Set.empty)
+    extends SimpleFeatureSerializer with LazyLogging {
+
+  private val kafkaAvroDeserializer = new KafkaAvroDeserializer(schemaRegistryClient)
+  private val wktReader = new WKTReader()
+  var geomSrcAttributeName: Option[String] = None
+
+  override def deserialize(id: String, bytes: Array[Byte], timestamp: Option[Long]): SimpleFeature = {
```

Comment:
I don't really like adding the timestamp to the serializer API just for this one use case - would it be possible to handle it in GeoMessageSerializer instead? So this would deserialize all the actual avro fields, and then GeoMessageSerializer could set the date based on the timestamp.